### PR TITLE
fix: Mark `opts` argument as optional

### PR DESCRIPTION
**Pull request by ExE-Boss**
This makes the `opts` argument be correctly marked as optional.

I’ve also fixed the `Index` type and made use of it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,12 +35,14 @@
...
     "@types/node": "*",
+    "@types/ssri": "^6.0.1",
     "decompress-maybe": "^1.0.0",
     "ssri": "^6.0.0",
     "tar-fs": "^2.0.0"
   },
   "devDependencies": {
-    "@types/tape": "^4.2.30",
+    "@types/tape": "^4.2.33",
+    "@types/tar-fs": "^1.16.1",
```
Comment (zkochan) [inline package.json]: this should be a prod dep

Comment (ExE-Boss) [reply package.json]: It doesn’t need to be a prod dependency, as none of the types from it are needed in the public type definition.

Unlike `ssri`, usage of `tar-fs` is merely an implementation detail.

**Comment by ExE-Boss** [comment]
needinfo?(@zkochan): Can we ship this?

**Comment by zkochan** [comment]
Unfortunately, the pnpm repo is not compiling with these changes. I need to look into it

**Comment by zkochan** [comment]
published as v6

**Pull request merged**